lang/src/gen/build.rs

```diff
@@ -48,6 +49,21 @@ pub fn generate_code(tokens: &mut TokenStream2, contract: &hir::Contract) {
     codegen_for_event_mod(tokens, contract);
 }
 
+fn codegen_for_env_types(tokens: &mut TokenStream2, contract: &hir::Contract) {
+    let env_types = &contract.env_types_type;
+    tokens.extend(quote! {
+        use ink_core::env::{self, EnvTypes, SrmlEnv};
+
+        type Env = SrmlEnv<#env_types>;
+        type EnvHandler = ink_model::EnvHandler<Env>;
```

Comment:
Why are you explicitely aliasing for these types?
Is it for the user or the later code gen?

Reply:
Both aliases used later in codegen. 

https://github.com/paritytech/ink/blob/33616695e6340e4579b939dbb2dfab298fb969d9/lang/src/gen/build.rs#L185

https://github.com/paritytech/ink/blob/33616695e6340e4579b939dbb2dfab298fb969d9/lang/src/gen/build.rs#L364

Alternatively we could generate the full type in each location with `SrmlEnv<#contract.env_types_type>`.

Though as observed in the previous comment the alias will surface to the user if they require the `EnvHandler` in a helper method.

Reply:
Okay I see. Maybe a good usability decision although in macro code I tend to always write out all names to make the generated code more robust in case of user inserted aliases or binding. Not sure if this code can be controlled in that way though.

Reply:
Replace type alias with explicit types in https://github.com/paritytech/ink/pull/108/commits/48732edf535f27825aed62d31a1d0eec4eafc998